packages/i18n/src/index.js

```diff
@@ -69,15 +57,13 @@ function getI18n() {
  *
  * @return {string} The translated string.
  */
-const dcnpgettext = memoize( ( domain = 'default', context, single, plural, number ) => {
-	try {
-		return getI18n().dcnpgettext( domain, context, single, plural, number );
-	} catch ( error ) {
-		logErrorOnce( 'Jed localization error: \n\n' + error.toString() );
-
-		return single;
+function dcnpgettext( domain = 'default', context, single, plural, number ) {
+	if ( ! i18n.data[ domain ] ) {
+		setLocaleData( undefined, domain );
 	}
-} );
+
+	return i18n.dcnpgettext( domain, context, single, plural, number );
```

Comment:
so I'm guessing this is to prevent errors when locale data for the domain has not been set yet?   I'm concerned about the potential for silent fails here.  Should we at least throw a development warning (a-la invariant or equivalent) when this is needed?

Reply:
> so I'm guessing this is to prevent errors when locale data for the domain has not been set yet? I'm concerned about the potential for silent fails here. Should we at least throw a development warning (a-la invariant or equivalent) when this is needed?

Related: https://github.com/WordPress/gutenberg/pull/11493#discussion_r231188718

I'll look into what the error paths would be and consider whether they're likely to be encountered and if there's a way to preempt / capture them.